internal/telemetry/client.go

```diff
@@ -174,24 +172,22 @@ func (c *client) Start(configuration []Configuration) {
 	c.metrics = make(map[Namespace]map[string]*metric)
 	c.debug = internal.BoolEnv("DD_INSTRUMENTATION_TELEMETRY_DEBUG", false)
 
+	productInfo := Products{
+		AppSec: ProductDetails{
+			Version: version.Tag,
+			Enabled: appsec.Enabled(),
+		},
+	}
+	productInfo.Profiler = ProductDetails{
+		Version: version.Tag,
+		// if the profiler is the one starting the telemetry client,
+		// then profiling is enabled
```

Comment:
What if it's the tracer starting the telemetry client, but profiling is enabled? Won't this say the profiler isn't enabled even though it could be?

Reply:
Yea - if the tracer starts the telemetry client first, profiling would be set as disabled in the initial `app-started` message. But then eventually when `profiler.Start` is called, an `app-product-change` event will be emitted to set profiling as enabled.